src/main/java/com/conveyal/datatools/manager/jobs/DeployJob.java

```diff
@@ -889,18 +959,57 @@ public String constructUserData(boolean graphAlreadyBuilt) {
                 manifest.runServer = false;
             } else {
                 // This instance will both run a graph and start the OTP server
-                manifest.routerConfigJSON = createEchoSafeJSON(deployment.generateRouterConfigAsString());
+                manifest.routerFolderDownloads.add(getRouterConfigS3Path());
+                if (
+                    !uploadStringToS3File(
+                        "router-config.json",
+                        deployment.generateRouterConfigAsString(),
+                        dryRun
+                    )
+                ) {
+                    return null;
+                }
+                routerConfigUploaded = true;
                 manifest.runServer = true;
                 manifest.uploadServerStartupLogs = true;
             }
         } else {
+            otpRunnerManifestS3Filename = "otp-runner-server-manifest.json";
             // This instance will only start the OTP server with a prebuilt graph
             manifest.buildGraph = false;
-            manifest.routerConfigJSON = createEchoSafeJSON(deployment.generateRouterConfigAsString());
+            manifest.routerFolderDownloads = Arrays.asList(getRouterConfigS3Path());
+            if (!routerConfigUploaded) {
+                if (
+                    !uploadStringToS3File(
+                        "router-config.json",
```

Comment:
This string and build-config.json are used multiple times and should probably be constants.